### PR TITLE
*: mitigations for Raft Engine encryption key issue

### LINKED ISSUE #12890: Encryption keys for WALs are not cleaned up when Raft Engine is enabled
### What version of TiKV are you using?

v6.1.0

### What operating system and CPU are you using?

N/A

### Steps to reproduce

Run write workload for a long time, observe the size of encryption key dictionary. The dictionary will grow indefinitely.

For a typical write workload, there are thousands of new WAL files generated by Raft Engine per day. The encryption keys associated with these files will not be cleaned up. They will occupy memory and disk space until one of them is full. Afterwards the server cannot be booted again.

This bug is not observed on any real cluster yet. Setting the severity to major for now.

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Close #12890

What's Changed:

- Update Raft Engine and implement new trait methods.
  - `delete` ensures encryption keys of new files can be properly cleaned up.
  - `delete_metadata` helps Raft Engine cleans up stale encryption keys left from older version of TiKV.
- Update tikv-ctl to support scanning and purging encryption keys. If the server is shutdown due to OOM or disk full, this operation can still be executed to free up space.

### Related changes

https://github.com/tikv/raft-engine/pull/229

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
Fix encryption keys not cleaned up when Raft Engine is enabled
```

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
Please fix the compile issue and update tikv-ctl doc

File: components/raft_log_engine/src/engine.rs [hunk 1]
```diff
@@ -174,14 +174,38 @@ impl FileSystem for ManagedFileSystem {
...
+                }
+            }
+        }
+        self.base_file_system.exists_metadata(path)
+    }
+
+    fn delete_metadata<P: AsRef<Path>>(&self, path: P) -> IoResult<bool> {
+        if let Some(ref manager) = self.key_manager {
+            // Note: no error if the file doesn't exist.
+            manager.delete_file(path.as_ref().to_str().unwrap())?
```
Comment (glorv) [inline components/raft_log_engine/src/engine.rs:202]:
```suggestion
            manager.delete_file(path.as_ref().to_str().unwrap())?;
```

**Review by glorv — COMMENTED** [review]
rest LGTM

**Comment by tabokie** [comment]
@Connor1996 It's blocked on https://github.com/tikv/raft-engine/pull/231

**Comment by tabokie** [comment]
Ping @Connor1996 @glorv

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by glorv — APPROVED** [review]

**Pull request merged**